File: app/src/main/java/com/vktiwari/pagingretrofitrxjava/network/NetworkState.java
package com.vktiwari.pagingretrofitrxjava.network;

public class NetworkState {
    public static final NetworkState LOADING;
    public static final NetworkState LOADED;

    public enum Status {
        RUNNING,
        SUCCESS,
        FAILED
    }

    private Status status;
    private String message;

    private NetworkState(Status status, String message) {
        this.status = status;
        this.message = message;
    }

    private NetworkState(Status status) {
        this.status = status;
    }

    static {
        LOADED = new NetworkState(Status.SUCCESS);
        LOADING = new NetworkState(Status.RUNNING,"Loading please wait");
    }

    public Status getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

    public static NetworkState error(String message) {
        return new NetworkState(Status.FAILED, message == null ? "unknown error" : message);
    }
}

File: app/src/kotlin/java/com/vktiwari/pagingretrofitrxjava/network/NetworkState.kt
package com.vktiwari.pagingretrofitrxjava.network

class NetworkState {

    enum class Status {
        RUNNING,
        SUCCESS,
        FAILED
    }

    private var status: Status? = null
    private var message: String?=null

    constructor(status: Status, message: String) {
        this.status = status
        this.message = message
    }

    constructor(status: Status){
        this.status = status
    }

    companion object {
        val LOADED: NetworkState = NetworkState(Status.SUCCESS);
        val LOADING = NetworkState(Status.RUNNING, "Loading please wait");

        fun error(message: String?): NetworkState {
            return NetworkState(Status.FAILED, message ?: "Something went wrong , try again")
        }
    }

    fun getStatus(): Status? {
        return status
    }

    fun getMessage(): String? {
        return message
    }

}